src/plugins/guided_onboarding/public/components/quit_guide_modal.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useState } from 'react';
+
+import {
+  EuiModal,
+  EuiModalBody,
+  EuiSpacer,
+  EuiTitle,
+  EuiText,
+  EuiModalFooter,
+  EuiButtonEmpty,
+  EuiButton,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import type { NotificationsSetup } from '@kbn/core-notifications-browser';
+import { GuideId } from '../../common/types';
+import { apiService } from '../services/api';
+
+interface QuitGuideModalProps {
+  closeModal: () => void;
+  currentGuide: GuideId;
+  notifications: NotificationsSetup;
+}
+
+export const QuitGuideModal = ({
+  closeModal,
+  currentGuide,
+  notifications,
+}: QuitGuideModalProps) => {
+  const [isDeleting, setIsDeleting] = useState<boolean>(false);
+
+  const deleteGuide = async () => {
+    setIsDeleting(true);
+    const { error } = await apiService.deleteGuide(currentGuide);
+
+    if (error) {
+      setIsDeleting(false);
+      notifications.toasts.addError(error, {
+        title: i18n.translate('guidedOnboarding.quitGuideModal.errorToastTitle', {
+          defaultMessage: 'There was an error quitting the guide. Please try again.',
+        }),
+      });
+    }
+    closeModal();
+  };
+
+  return (
+    <EuiModal
+      maxWidth={448}
+      aria-label="quitGuideModal"
+      onClose={closeModal}
+      data-test-subj="quitGuideModal"
+    >
+      <EuiModalBody>
+        <EuiSpacer size="m" />
+        <EuiTitle size="m">
+          <h2>
+            {i18n.translate('guidedOnboarding.quitGuideModal.modalTitle', {
+              defaultMessage: 'Quit this setup guide and discard progress?',
```

Comment:
```suggestion
              defaultMessage: 'Quit this guide and discard progress?',
```
I think we can tweak this a bit more, since they already know they're in a setup guide.